app/controllers/auth_controller.rb

```diff
@@ -1,6 +1,8 @@
 class AuthController < ApplicationController
   helper :auth
 
+  @@attempts = 0
```

Comment:
I got another idea which might be better than using a class variable. in your PasswordReset model, you can have a column "create_at". When a user request to reset the password, the code can check the # of reset requests done by the same user in, say, 24 hours. If the number is >=3, then doing recaptcha test is required.